### ISSUE TITLE
Bug in overflow multiplication of uint type.

### ISSUE THREAD

**Issue by shurcooL** [gopherjs/gopherjs#652]
This might affect `int` type too, I haven't verified.

It does _not_ affect `uintptr`, nor `uint32`.

```Go
package main

import (
	"fmt"
)

func main() {
	x := uint(4294967295)
	fmt.Println(x * x)
	// Output (gc):       1
	// Output (GopherJS): 0
}
```

https://play.golang.org/p/gY-qQ51AyZ
https://gopherjs.github.io/playground/#/gY-qQ51AyZ

/cc @neelance

I found this because `math/big` changed type of `big.Word` from `uintptr` in Go 1.8 to `uint` in Go 1.9, and it broke the majority of its tests.

**Comment by flimzy** [comment]
My local testing still shows an inconsistency between Go and GopherJS:

```
jonhall@ivy:~/src/test$ cat main.go
package main

import (
        "fmt"
)

func main() {
        x := uint(4294967295)
        fmt.Println(x * x)
        // Output (gc):       1
        // Output (GopherJS): 0
}
jonhall@ivy:~/src/test$ gopherjs run main.go
0
jonhall@ivy:~/src/test$ gopherjs version
GopherJS 1.19.0-beta1+go1.19.13
jonhall@ivy:~/src/test$ go run .
18446744065119617025
jonhall@ivy:~/src/test$ go version
go version go1.24.4 linux/amd64
jonhall@ivy:~/src/test$ 
```

(Yes, different Go version than GopherJS, but I doubt that should matter for multiplication 😉 ).  Not sure why @dmitshur saw an output of `1` from Go.  Maybe he was testing on a 32-bit system?

In any case, GopherJS's arithmetic still seems off in this case.

**Comment by dmitshur** [comment]
> Not sure why @dmitshur saw an output of `1` from Go. Maybe he was testing on a 32-bit system?

Yes, I think so. I should've written it down explicitly, but it makes sense to compare gc behavior on 32-bit ports since GopherJS's `uint` is [32-bit](https://github.com/gopherjs/gopherjs?tab=readme-ov-file#architecture).
